### PR TITLE
Look for MeCab dictionaries in various places besides this package

### PR DESCRIPTION
**Pull request by rspeer**
MeCab dictionaries are too large for PyPI, so we need to be able to leave them out of the wordfreq package and still find them elsewhere.

### CODE REVIEW

**Comment by rspeer** [comment]
Now that I know MeCab can't use filenames with long paths, that means that absolutely nothing is helped by including the MeCab files within this repository. I've just put directions on how to install MeCab dictionaries in the README instead.

File: wordfreq/mecab.py [hunk 1]
```diff
@@ -1,12 +1,65 @@
...
+
+
+# MeCab has fixed-sized buffers for many things, including the dictionary path
+MAX_PATH_LENGTH = 58
+
+
+def find_mecab_dictionary(names):
+    """
+    Find a MeCab dictionary with a given name. The dictionary might come as
+    part of this repository (if you got wordfreq from GitHub) or might have to
```
Comment (alin-luminoso) [inline wordfreq/mecab.py]: Is this first part still true?

File: wordfreq/mecab.py [hunk 2]
```diff
@@ -1,12 +1,65 @@
...
+    be installed separately (if you got wordfreq from PyPI).
+
+    We'd prefer to include MeCab in the repository all the time, but PyPI's
+    package size limits make that not an option.
```
Comment (alin-luminoso) [inline wordfreq/mecab.py]: (Ditto.)

**Comment by rspeer** [comment]
I think this is ready to be re-reviewed.

File: wordfreq/mecab.py [hunk 3]
```diff
@@ -1,12 +1,61 @@
...
+    """
+    suggested_pkg = names[0]
+    paths = [
+        os.path.expanduser('~/.local/lib/mecab/dic'),
+        '/var/lib/mecab/dic',
+        '/var/local/lib/mecab/dic',
+        '/usr/lib/mecab/dic',
+        '/usr/local/lib/mecab/dic',
+    ]
+    full_paths = [os.path.join(path, name) for path in paths for name in names]
```
Comment (alin-luminoso) [inline wordfreq/mecab.py]: Would it be worth consolidating the `MAX_PATH_LENGTH` logic?  For example, compute `skipped_paths` and `checked_paths` up here, or something.

File: wordfreq/mecab.py [hunk 4]
```diff
@@ -1,12 +1,61 @@
...
+    for path in full_paths:
+        if os.path.exists(path) and len(path) <= MAX_PATH_LENGTH:
+            return path
+
+    error_lines = [
+        "Couldn't find the MeCab dictionary named %r." % suggested_pkg,
+        "You should download or use your system's package manager to install",
+        "the %r package." % suggested_pkg,
+        "",
+        "We looked in the following locations:"
```
Comment (alin-luminoso) [inline wordfreq/mecab.py]: Nitpick: might read better for there to be a space after this colon.

File: wordfreq/mecab.py [hunk 5]
```diff
@@ -1,12 +1,61 @@
...
+
+    raise OSError('\n'.join(error_lines))
+
+
+def make_mecab_analyzer(names):
+    """
+    Get a MeCab analyzer object, given a list of names the dictionary might
+    have.
+    """
+    filename = find_mecab_dictionary(names)
```
Comment (alin-luminoso) [inline wordfreq/mecab.py]: Nitpick: is this temporary useful?

**Comment by alin-luminoso** [comment]
Okay, I had some nitpicks, but that's all.

**Comment by rspeer** [comment]
Okay, I think I've handled those.

**Comment by alin-luminoso** [comment]
Okay, I think this is ready for merging as soon as the Jenkins workers have the files they need.

**Pull request merged**